STMicroelectronics/STM32L4_L4+/app/az_ulib_dm/src/az_ulib_registry.c

```diff
@@ -0,0 +1,199 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
+
+#include "_az_ulib_registry_interface.h"
+#include "az_ulib_ipc_api.h"
+#include "az_ulib_ipc_interface.h"
+#include "az_ulib_registry_api.h"
+#include "az_ulib_result.h"
+#include "stm32l475_flash_driver.h"
+#include <azure/core/internal/az_precondition_internal.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <string.h>
+
+#define MAX_AZ_ULIB_REGISTRY_ENTRIES 10
+
+static az_registry_block registry_editor[MAX_AZ_ULIB_REGISTRY_ENTRIES];
+
+extern uint32_t __REGISTRYINFO;
+extern uint32_t __REGISTRY;
+
+static az_result result_from_hal_status(HAL_StatusTypeDef status)
+{
+  switch (status)
+  {
+    case HAL_OK:
+      return AZ_OK;
+    case HAL_ERROR:
+      return AZ_ERROR_ULIB_SYSTEM;
+    case HAL_BUSY:
+      return AZ_ERROR_ULIB_BUSY;
+    case HAL_TIMEOUT:
+      return AZ_ERROR_ULIB_SYSTEM; // should be AZ_ERROR_ULIB_TIME_OUT
+    default:
+      return AZ_ERROR_ULIB_SYSTEM;
+  }
+}
+
+static az_result store_registry()
+{
+  AZ_ULIB_TRY
+  {
+    az_result result;
+    HAL_StatusTypeDef hal_status;
+
+    int32_t registry_size = sizeof(az_registry_block) * MAX_AZ_ULIB_REGISTRY_ENTRIES;
+
+    // erase registry
+    if ((hal_status = internal_flash_erase((uint8_t*)(&__REGISTRYINFO), registry_size)) != HAL_OK)
```

Comment:
>internal_flash_erase [](http://example.com/codeflow?start=22&length=20)

You cannot erase the flash to write a new register.

Reply:
Let's chat offline about it.

Reply:
Will no longer erase before writing new node.